bundles/org.openhab.binding.epsonprojector/src/main/java/org/openhab/binding/epsonprojector/internal/connector/EpsonProjectorSerialConnector.java

```diff
@@ -0,0 +1,209 @@
+            throw new EpsonProjectorException(e);
+        }
+    }
+
+    @Override
+    public void disconnect() throws EpsonProjectorException {
+        InputStream in = this.in;
+        OutputStream out = this.out;
+        SerialPort serialPort = this.serialPort;
+
+        if (out != null) {
+            logger.debug("Close serial out stream");
+            try {
+                out.close();
+            } catch (IOException e) {
+                logger.debug("Error occurred when closing serial out stream: {}", e.getMessage());
+            }
+            this.out = null;
+        }
+        if (in != null) {
+            logger.debug("Close serial in stream");
+            try {
+                in.close();
+            } catch (IOException e) {
+                logger.debug("Error occurred when closing serial in stream: {}", e.getMessage());
+            }
+            this.in = null;
+        }
+        if (serialPort != null) {
+            logger.debug("Close serial port");
+            serialPort.close();
+            serialPort.removeEventListener();
+            this.serialPort = null;
+        }
+
+        logger.debug("Closed");
+    }
+
+    @Override
+    public String sendMessage(String data, int timeout) throws EpsonProjectorException {
+        InputStream in = this.in;
+        OutputStream out = this.out;
+
+        if (in == null || out == null) {
+            connect();
+            in = this.in;
+            out = this.out;
+        }
+
+        try {
+            if (in != null && out != null) {
+                // flush input stream
+                if (in.markSupported()) {
+                    in.reset();
+                } else {
+                    while (in.available() > 0) {
+                        int availableBytes = in.available();
+
+                        if (availableBytes > 0) {
+                            byte[] tmpData = new byte[availableBytes];
+                            in.read(tmpData, 0, availableBytes);
+                        }
+                    }
+                }
+                return sendMmsg(data, timeout);
+            } else {
+                return "";
+            }
+        } catch (IOException e) {
+            logger.debug("IO error occurred...reconnect and resend once: {}", e.getMessage());
+            disconnect();
+            connect();
+
+            try {
+                return sendMmsg(data, timeout);
+            } catch (IOException e1) {
+                throw new EpsonProjectorException(e);
+            }
+        }
+    }
+
+    @Override
+    public void serialEvent(SerialPortEvent arg0) {
+    }
+
+    private String sendMmsg(String data, int timeout) throws IOException, EpsonProjectorException {
+        String resp = "";
+
+        InputStream in = this.in;
+        OutputStream out = this.out;
+
+        if (in != null && out != null) {
+            out.write(data.getBytes());
+            out.write("\r\n".getBytes());
```

Comment:
The default charset varies from system to system so it is better to specify it yourself instead of using the default.

Reply:
fixed